src/lib/datatypes/include/sol-buffer.h

```diff
@@ -420,6 +420,17 @@ int sol_buffer_append_slice(struct sol_buffer *buf, const struct sol_str_slice s
 int sol_buffer_append_bytes(struct sol_buffer *buf, const uint8_t *bytes, size_t size);
 
 /**
+ * @brief Appends the contents of the buffer @c from in the end of @c buf,
+ * reallocating @c if necessary.
+ *
+ * @param dst Destination buffer
+ * @param src The buffer from where the data will be copied
+ *
+ * @return @c 0 on success, error code (always negative) otherwise
+ */
+int sol_buffer_append_buffer(struct sol_buffer *dst, const struct sol_buffer *src);
```

Comment:
ok, still good... but I'd make all of these single-line helpers static inlines... so we don't have to maintain many symbols in our ABI.